iron-focusables-helper.html

```diff
@@ -0,0 +1,135 @@
+      },
+
+      /**
+       * Returns if a node is focusable.
+       * @param {!Node} node
+       * @return {boolean}
+       */
+      isFocusable: function(node) {
+        // shadow roots won't have this method.
+        if (!node.hasAttribute) {
+          return false;
+        }
+        var name = node.localName;
+        // Elements that can be focused even if they have [disabled] attribute.
+        if (name === 'iframe' ||
+          ((name === 'a' || name === 'area') && node.hasAttribute('href')) ||
+          node.hasAttribute('tabindex') || node.hasAttribute('contentEditable')) {
+          return true;
+        }
+        // Elements that cannot be focused if they have [disabled] attribute.
+        return /^(input|select|textarea|button|object)$/.test(name) && !node.disabled;
+      },
+
+      /**
+       * Returns if a node is tabbable.
+       * @param {!Node} node
+       * @return {boolean}
+       */
+      isTabbable: function(node) {
+        return node.tabIndex >= 0 && this.isFocusable(node) && this._isVisible(node);
+      },
+
+      /**
+       * Searches for nodes that are tabbable and adds them to the `result` array.
+       * Returns the maximum tabindex of the added nodes.
+       * @param {!Node} node The starting point for the search; it's added if tabbable.
+       * @param {!Array<Node>} result
+       * @return {Number}
+       * @private
+       */
+      _findTabbableNodes: function(node, result) {
+        var maxTabIndex = 0;
+        // Skip #text nodes. If not visible, no need to explore children.
+        if (node.nodeType !== 3 && this._isVisible(node)) {
+          if (this.isTabbable(node)) {
+            result.push(node);
+            maxTabIndex = node.tabIndex;
+          }
+          var children;
+          if (node.localName === 'content') {
+            children = Polymer.dom(node).getDistributedNodes();
+          } else {
+            // Use shadow root if possible, will check for distributed nodes.
+            children = Polymer.dom(node.root || node).children;
+          }
+          for (var i = 0; i < children.length; i++) {
+            maxTabIndex = Math.max(this._findTabbableNodes(children[i], result), maxTabIndex);
+          }
+        }
+        return maxTabIndex;
+      },
+
+      /**
+       * Returns false if the node has `visibility: hidden` or `display: none`
+       * @param {!Node} node
+       * @return {boolean}
+       * @private
+       */
+      _isVisible: function(node) {
+        // Check inline style first, might save a re-flow.
+        return this._isStyleVisible(node.style) && this._isStyleVisible(window.getComputedStyle(node));
+      },
+
+      /**
+       * Returns false if the style is `visibility: hidden` or `display: none`
+       * @param {!CSSStyleDeclaration} style
+       * @return {boolean}
+       * @private
+       */
+      _isStyleVisible: function(style) {
+        return (style.visibility !== 'hidden' && style.display !== 'none');
```

Comment:
I think you'll need to test if `getBoundingClientRect` returns all zeroes like you mentioned before, rather than checking for `display: none;`: http://jsbin.com/moyoma/1/edit?html,output